src/core/src/service_interfaces/TelemetryWriter.py

```diff
@@ -164,70 +169,84 @@ def write_event(self, message, event_level=Constants.TelemetryEventLevel.Informa
+                raise
+
+        except Exception as e:
+            self.composite_logger.log_telemetry_module_error("Error occurred while deleting older telemetry events. [Error={0}]".format(repr(e)))
             raise
 
     def __get_file_and_content_to_write(self, folder_path, data):
-        """ Identifies the file where the event is to be written can be an existing event file or a new one depending upon the size restrictions. If event is to be written to an existing file, fetches retains it's content """
-        file_path = self.__get_event_file_path(folder_path)
-        all_events = []
-        if os.path.exists(file_path):
-            file_size = self.get_file_size(file_path)
-            # if file_size exceeds max limit, sleep for 1 second, so the event can be written to a new file since the event file name is a timestamp
-            if file_size >= Constants.TELEMETRY_EVENT_FILE_SIZE_LIMIT_IN_CHARS:
-                time.sleep(1)
-                file_path = self.__get_event_file_path(folder_path)
-            else:
-                all_events = self.__fetch_events_from_previous_file(file_path)  # fetches existing content within the file
-        all_events.append(data)
-        return file_path, all_events
+        """ Identifies the file where the event is to be written. Can be an existing event file or a new one depending upon the size restrictions. If event is to be written to an existing file, fetches retains it's content """
+        try:
+            file_path = self.__get_event_file_path(folder_path)
+            all_events = []
+            if os.path.exists(file_path):
+                file_size = self.get_file_size(file_path)
+                # if file_size exceeds max limit, sleep for 1 second, so the event can be written to a new file since the event file name is a timestamp
+                if file_size >= Constants.TELEMETRY_EVENT_FILE_SIZE_LIMIT_IN_CHARS:
+                    time.sleep(1)
+                    file_path = self.__get_event_file_path(folder_path)
+                else:
+                    all_events = self.__fetch_events_from_previous_file(file_path)  # fetches existing content within the file
+            all_events.append(data)
+            return file_path, all_events
+        except Exception as e:
+            self.composite_logger.log_telemetry_module_error("Error occurred while fetching event file to write the event to. [Error={0}]")
+            raise
 
     def __throttle_telemetry_writes_if_required(self, is_event_file_throttling_needed=True):
         """ Ensures the # of event files that can be written per time unit restriction is met. Returns False if the any updates are required after the restriction enforcement. For eg: file_name is a timestamp and should be modified if a wait is added here """
```

Comment:
For future callers, document in a comment here who exactly is allowed to call with the throttling disabled and why.

Reply:
This was added in write_event, which passes this arg down to __throttle_telemetry_writes_if_required

line 152-153:
def write_event(self, message, event_level=Constants.TelemetryEventLevel.Informational, task_name=Constants.TELEMETRY_TASK_NAME, is_event_file_throttling_needed=True):
        """ Creates and writes event to event file after validating none of the telemetry size restrictions are breached
        NOTE: is_event_file_throttling_needed is used to determine if event file throttling is required and as such should always be True.
        The only scenario where this is False is when throttling is taking place and we write to telemetry about it. i.e. only from within __throttle_telemetry_writes_if_required()"""

Reply:
Added a similar comment, as write_event(), in __throttle_telemetry_writes_if_required()